backend/search/base/controller.go

```diff
@@ -0,0 +1,75 @@
+package base
+
+import (
+	"net/http"
+
+	search_types "github.com/GenerateNU/sac/backend/search/types"
+	"github.com/GenerateNU/sac/backend/utilities"
+	"github.com/gofiber/fiber/v2"
+)
+
+type SearchController struct {
+	searchService SearchServiceInterface
+}
+
+func NewSearchController(searchService SearchServiceInterface) *SearchController {
+	return &SearchController{searchService: searchService}
+}
+
+// SearchClubs godoc
+//
+// @Summary		Searches through clubs
+// @Description	Searches through clubs
+// @ID			search-club
+// @Tags      	search
+// @Accept		json
+// @Produce		json
+// @Param		searchQuery	query	models.SearchQueryParams	true	"Search Body"
+// @Success		200	  {object}	    []models.ClubSearchResult
+// @Failure     404   {object}		error
+// @Failure     500   {object}      error
+// @Router		/search/clubs  [get]
+func (s *SearchController) SearchClubs(c *fiber.Ctx) error {
```

Comment:
does this need to handle pagination? does opensearch support it?

Reply:
From experiimenting with it queries are only returning at most up to 10 results, I dont see why pagination would be necessary since thats a small amount

Reply:
sounds good 👍